File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/CallState.java
package com.apollographql.apollo.internal;

enum CallState {
  IDLE, ACTIVE, TERMINATED, CANCELED;

  static class IllegalStateMessage {
    private final CallState callState;

    private IllegalStateMessage(CallState callState) {
      this.callState = callState;
    }

    static IllegalStateMessage forCurrentState(CallState callState) {
      return new IllegalStateMessage(callState);
    }

    String expected(CallState... acceptableStates) {
      StringBuilder stringBuilder = new StringBuilder("Expected: " + callState.name() + ", but found [");
      String deliminator = "";
      for (CallState state : acceptableStates) {
        stringBuilder.append(deliminator).append(state.name());
        deliminator = ", ";
      }
      return stringBuilder.append("]").toString();
    }
  }
}

File: deprecated-apollo-runtime/src/main/java/com/apollographql/apollo3/internal/CallState.kt
package com.apollographql.apollo3.internal

enum class CallState {
  IDLE, ACTIVE, TERMINATED, CANCELED;

  internal class IllegalStateMessage private constructor(private val callState: CallState) {
    fun expected(vararg acceptableStates: CallState): String {
      val stringBuilder = StringBuilder("Found: " + callState.name + ", but expected [")
      var deliminator = ""
      for (state in acceptableStates) {
        stringBuilder.append(deliminator).append(state.name)
        deliminator = ", "
      }
      return stringBuilder.append("]").toString()
    }

    companion object {
      @JvmStatic
      fun forCurrentState(callState: CallState): IllegalStateMessage {
        return IllegalStateMessage(callState)
      }
    }
  }
}